packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx

```diff
@@ -135,6 +135,31 @@ function CategoryList({
   );
 }
 
+type Suggestion = {
+  id: string;
+  name: string;
+  group?: {
+    name: string;
+  };
+};
+
+function customSort(obj: Suggestion, value: string): number {
+  const name = obj.name.toLowerCase();
+  const groupName = obj.group ? obj.group.name.toLowerCase() : '';
+  value = value.toLowerCase();
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏we should treat variables as immutable. Otherwise this has the potential to introduce subtle and hard-to-debug bugs.

```suggestion
  const lowerCaseValue = value.toLowerCase();
```

Reply:
Is this even needed? The calls to `customSort` below seem to pass in lowercased values anyway.

Reply:
A safer approach (in case someone adds more calls to this function) would be to lowercase in here as it does, and just drop the lowercase at the call sites below.